web/blog-template/web/components/PostForm.tsx

```diff
@@ -83,18 +81,18 @@ export default ({ postToEdit }) => {
   }, [postToEdit]);
 
   // submit the form
-  const handleSave = async (event) => {
+  const handleSave = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
 
     // replace all newlines with <br /> tags for markdown compatibility
     const content = getValues("content");
-    if (!content.markdown) {
-      content.replaceAll("\n", "<br />");
+    if (typeof content == "string") {
+      content?.replace(/\n/g, "<br />");
 
       // reformat the content field to be a markdown field
       setValue("content", {
         markdown: content,
-      });
+      } as any);
```

Comment:
Instead of doing this, we should be able to type the default value (line 40 of this file)

```
content: "Start writing your blog post here!" as string | { markdown: string | (string & { markdown: string | null | undefined; }); },
```

Reply:
I'll also try to find a big of time to clean this up before we pivot outside of Shopify apps, it is an icky MDXEditor integration right now

Reply:
I tried that type but it didn't seem to work. Not sure what to do in this case. Do you have an suggestions?

Reply:
Did you try it on line 40? Seems to work okay here:
https://blog-ts-template-test.gadget.app/edit/development/files/web/components/PostForm.tsx?startLineNumber=40&startColumn=0&endLineNumber=40&endColumn=0

Reply:
Ok, I just realized what you meant here. Its fixed!